crates/burn-jit/src/ops/base.rs

```diff
@@ -1,36 +1,37 @@
 use crate::{element::JitElement, kernel, tensor::JitTensor, JitRuntime};
 use burn_cube::CubeElement;
-use burn_tensor::{Data, Reader, Shape};
+use burn_tensor::{Reader, Shape, TensorData};
 use std::marker::PhantomData;
 
 pub(crate) fn from_data<R: JitRuntime, E: JitElement, const D: usize>(
-    data: Data<E, D>,
+    data: TensorData,
     device: &R::Device,
 ) -> JitTensor<R, E, D> {
+    let shape: Shape<D> = (&data.shape).into();
     let client = R::client(device);
-    let buffer = client.create(E::as_bytes(&data.value));
+    let buffer = client.create(E::as_bytes(data.convert::<E>().as_slice().unwrap()));
 
```

Comment:
I would create a function `as_bytes` directly on `TensorData` to avoid the double casting here.

```rust
client.create(data.convert::<E>().as_bytes());
```